### ISSUE TITLE
Change validation_step to scale back

### ISSUE THREAD

**Issue by kdgutier** [Nixtla/neuralforecast#319]
Currently `validation_step` is tracking the validation loss mimicking the training procedure's.

This generates two problems:
- It reduces the correlation between the validation and the scaled test signal.
- The automatic hyper parameter optimizer will trivially select the `scale_type` that minimizes val_loss.
